TypeCobol/Compiler/CupPreprocessor/CompilerDirectiveBuilder.cs

```diff
@@ -241,41 +241,73 @@ public void EnterSequenceNumberField(List<Token> seqNumField)
 
-            bool isFirst = true;
-            int previous = -42;
-            foreach(Token integerLiteralToken in seqNumField)
-            {
-                System.Diagnostics.Debug.Assert(integerLiteralToken.TokenType == TokenType.IntegerLiteral);
-                int current = (int)((IntegerLiteralTokenValue)integerLiteralToken.LiteralValue).Number;
 
-                if (isFirst)
+            foreach (Token token in seqNumField)
+            {
+                if (token.TokenType == TokenType.IntegerLiteral)
                 {
-                    previous = current;
-                    isFirst = false;
+                    int current = (int)((IntegerLiteralTokenValue)token.LiteralValue).Number;
+
+                    //Report error for negative number and ignore them
+                    if (current < 0)
+                    {
+                        AddInvalidIntegerDiagnostic(current.ToString());
+                    }
+                    else
+                    {
+                        DeleteDirective.SequenceNumberRange range = new DeleteDirective.SequenceNumberRange();
+                        range.From = current;
+                        range.To = current;
+                        deleteDirective.SequenceNumberRangesList.Add(range);
+                    }
                 }
-                else
+                else if (token.TokenType == TokenType.UserDefinedWord)
                 {
-                    DeleteDirective.SequenceNumberRange range = new DeleteDirective.SequenceNumberRange();
-                    range.From = previous;
-                    if (current < 0)
+                    var numbers = token.Text.Split('-');
+
+                    if (numbers.Length != 2)
                     {
-                        range.To = -current;
-                        isFirst = true;
+                        Diagnostic error = new Diagnostic(MessageCode.SyntaxErrorInParser, token.Position(),
+                            "Invalid range format");
+                        CompilerDirective.AddDiagnostic(error);
                     }
                     else
                     {
-                        range.To = previous;
-                        previous = current;
+                        DeleteDirective.SequenceNumberRange range = new DeleteDirective.SequenceNumberRange();
+                        if (TryParseIntOrAddDiagnostic(numbers[0], out var temp))
+                        {
+                            range.From = temp;
+                            if (TryParseIntOrAddDiagnostic(numbers[1], out temp))
+                            {
+                                range.To = temp;
+                                deleteDirective.SequenceNumberRangesList.Add(range);
+                            }
+                        } //else ignore values
+
+                        bool TryParseIntOrAddDiagnostic(string invalidInteger, out int result)
```

Comment:
'invalidInteger' ? It may be a valid integer text otherwise the method is pointless. Do you mean 'integerText' or 'numberText' ?